android/app/src/main/java/com/bitcoin/assetconverter/providers/ProviderAssetManager.kt

```diff
@@ -0,0 +1,34 @@
+package com.bitcoin.assetconverter.providers
+
+import android.content.Context
+import com.bitcoin.assetconverter.models.ProviderAssets
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+
+class ProviderAssetManager{
+
+    val assets: MutableMap<ProviderType, List<ProviderAssets.Assets>> = mutableMapOf()
+    private var initialized = false
```

Comment:
we can use `lateinit var` to reduce memory allocation.

Reply:
This is not allowed: 

```
lateinit' modifier is not allowed on properties of primitive types
```